core/lib/utils.ts

```diff
@@ -1,2 +1,20 @@
 export const noop = () => {};
 export const identity = <T>(x: T) => x;
+
+/**
+ * @param number - decimal number
+ * @returns the decimal precision of the number
+ */
+export function getDecimalPrecision(number: number) {
+	const regexp = ('' + number).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/);
```

Comment:
Just a little point here, but I think it's usually better to create the regexp only once, so define it outside the function. It avoid the recompilation of the regexp pattern.
```
const decimalRegExp = /(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/;
```

then (I would prefer another name for regexp, as it's the result of the match, not the regexp itself)
```
const matches = ('' + number).match(decimalRegExp);
```